cmd/aro/deploy.go

```diff
@@ -15,49 +15,73 @@ import (
 	deployer "github.com/Azure/ARO-RP/pkg/deploy"
 )
 
-func deploy(ctx context.Context, log *logrus.Entry) error {
-	deployVersion, location := gitCommit, flag.Arg(2)
-
-	if os.Getenv("RP_VERSION") != "" {
-		deployVersion = os.Getenv("RP_VERSION")
-	}
-
-	log.Printf("deploying version %s to location %s", deployVersion, location)
-
-	if deployVersion == "unknown" || strings.Contains(deployVersion, "dirty") {
-		return fmt.Errorf("invalid deploy version %q", deployVersion)
-	}
+type flags struct {
+	configFile      string
+	deployVersion   string
+	location        string
+	isPreDeployOnly bool
```

Comment:
should we merge fullDeploy and isPreDeployOnly and make it a parameter?  e.g. DEPLOY_MODE := {predeploy,deploy,upgrade} ?

Reply:
Dam! This was my first version and I removed it for lower number of configurables :D

Reply:
In this case, it will be `Pre-Deploy, Deploy, and Upgrade (which is Deploy + Upgrade)`

If we want to go this path I think we better doing this:
`deploy ./condig.yaml {predeploy,deploy,upgrade} eastus` and avoid "one more env variable?

Reply:
yes